proto/viam/app/v1/app.proto

```diff
@@ -895,7 +895,7 @@ message ListMachineFragmentsRequest {
 
 message ListMachineFragmentsResponse {
   repeated Fragment fragments = 1;
-  repeated FragmentTree fragment_trees = 2;
+  repeated FragmentWithError resolved_fragments = 2;
```

Comment:
fyi, this has some implications for nested fragments and fragment overwrites behavior. if we return the resolved fragments (the resultant config after all nesting is applied), fragment overwrites will operate on the resolved fragment as a whole, rather than on each nested fragment within a fragment tree.

here is an example. given this fragment tree:
```
a
|
b
|
c
```
if i want to apply a fragment overwrite to fragment b (nested within fragment a), i add a fragment mod to fragment a in my part config.

the alternative to this is to have fragment mods apply to nested fragments. so, given the same fragment tree, you would include a fragment mod on fragment b in the part config. this would allow you to resolve any conflicts between nested fragments (like fragments b and c) before they are merged. however, i think this approach is confusing to use because it requires the user to understand the nested structure of the fragment, and this would require us to push more duplicate computation on the frontend. @mcous and i spoke and we think it's okay to push nested-fragment-resolution to a later project (ie. fragment overwrites on fragments). 

please leave any thoughts you have on this!

Reply:
great i agree that pushing the resolution to a later project works as long as this decision isnt going to make that _more_ difficult. this current logic still makes sense to me too